grepos/app/src/main/java/net/boondockradio/grepos/adapter/RepositoryAdpter.java

```diff
@@ -3,47 +3,121 @@
+public class RepositoryAdpter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    private final int VIEW_ITEM = 1;
+    private final int VIEW_PROGRESS = 0;
+
+    private final int LOAD_MORE_THRESHOLD = 2;
 
     private List<Repository> mRepositoryItems;
+    private RecyclerView mRecyclerView;
+
+    private OnLoadMoreListener onLoadMoreListener;
+    private int total = 0;
+    private int last = 0;
+    private boolean isLoading = false;
 
-    public RepositoryAdpter(List<Repository> repositoryItems) {
+    public RepositoryAdpter(List<Repository> repositoryItems, RecyclerView recyclerView) {
         mRepositoryItems = repositoryItems;
+        mRecyclerView = recyclerView;
+
+        final LinearLayoutManager layoutManager = (LinearLayoutManager) mRecyclerView.getLayoutManager();
+        mRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                total = layoutManager.getItemCount();
+                last = layoutManager.findLastVisibleItemPosition();
+
+                if (dy < 0) {
+                    return;
+                }
+
+                if (!isLoading && last + LOAD_MORE_THRESHOLD >= total) {
+                    onLoadMoreListener.onLoadMore();
+                    isLoading = true;
+                }
+            }
+        });
     }
 
     @Override
-    public ItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-        View v = LayoutInflater
-                .from(parent.getContext())
-                .inflate(R.layout.raw_item, parent, false);
-        return new ItemViewHolder(v);
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        RecyclerView.ViewHolder vh;
+        View v;
+
+        switch (viewType) {
+            case VIEW_ITEM:
+                v = LayoutInflater
+                        .from(parent.getContext())
+                        .inflate(R.layout.row_item, parent, false);
+                vh = new ItemViewHolder(v);
+                break;
+
+            default:
+                v = LayoutInflater
+                        .from(parent.getContext())
+                        .inflate(R.layout.row_progress, parent, false);
+                vh = new ProgressViewHolder(v);
+        }
+
+        return vh;
     }
 
     @Override
-    public void onBindViewHolder(ItemViewHolder holder, int position) {
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
         Repository item = mRepositoryItems.get(position);
-        holder.textView.setText(item.fullName);
+        if (holder instanceof ItemViewHolder) {
+            ((ItemViewHolder) holder).textView.setText(item.fullName);
+        }
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        return mRepositoryItems.get(position) != null ? VIEW_ITEM : VIEW_PROGRESS;
```

Comment:
personally I dont like `null`. `null` checks can make code difficult to read and maintain.

You can do this to avoid the need for null checks;
``` java
private List<Repository> mRepositoryItems;

@Override
public int getItemViewType(int position) {
       return position == mRepositoryItems.size() ? VIEW_PROGRESS : VIEW_ITEM;
}
```